tests/integration/iop-integration-test-defaults.yaml

```diff
@@ -5,6 +5,9 @@ spec:
   meshConfig:
     accessLogFile: "/dev/stdout"
   components:
+    ingressGateways:
+    - name: istio-ingressgateway
```

Comment:
`ingressGateways` and `meshExpansionPorts` sections needed to be initialized with an array otherwise the merge of the two top files breaks the ability to use `--set` on array lists in the integration framework
```
	installSettings := []string{
		"-f", defaultsIOPFile,
		"-f", iopFile,
```
for instance, `istioctl manifest generate --set components.ingressGateways[0].name=istio-ingressgateway` works as expected, but in test framework `installSettings = append(installSettings, "--set", "components.ingressGateways[0].name=istio-ingressgateway")` results in an error
```
2020-06-01T20:07:32.638174Z	error	tf	Exiting due to setup failure: failed deploying control plane to cluster 1: failed to deploy centralIstiod base for cluster cluster-1: could not unmarshal merged YAML: json: cannot unmarshal object into Go value of type []json.RawMessage
...
...
  ingressGateways:
    '[0]':
      enabled: false
```
cc @esnible @ostromart

Reply:
created issue for tracking purposes:
https://github.com/istio/istio/issues/24377